### ISSUE TITLE
Problem with jsx-closing-bracket-location rule

### ISSUE THREAD

**Issue by playpauseandstop** [yannickcr/eslint-plugin-react#208]
After enabling

```
    "react/jsx-closing-bracket-location": [2, {"location": "after-props"}],
```

rule in project's ESLint config, next element start producing errors,

```jsx
<Button
    bsStyle="success"
    data-id={showPromoteModal}
    onClick={this.handleUserPromote}>
    Promote
</Button>
```

But from rule definition I believe that it should only validate self-closing JSX tags.

I'm not sure how better handle this situation, maybe adding some configuration value to the rule, but in total, when rule enabled I only want to validate closing bracket location for self-closing JSX tags, like,

```jsx
<Route
    name="page"
    handler={Page}
    path="/page/:id" />
```

and do not validate normal tags, like `Button` above.

What do you think? Does it make sense for you?

**Comment by yannickcr** [comment]
Well, currently the rule enforce the same style for self-closing and non-self-closing tags and do not make any distinction. 

What you want is to be able to configure two different style for self-closing and non-self-closing tags, or enforce only one ?

**Comment by playpauseandstop** [comment]
Yep, I want be able to configure two different styles for self-closing and non-self-closing tags.

**Comment by yannickcr** [comment]
Ok, it could be a future option for this rule.

**Comment by evocateur** [comment]
+1 for splitting them up.
